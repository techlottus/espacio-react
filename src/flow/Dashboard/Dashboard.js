import React, { useEffect, useState } from "react";

import CardHeader from "../../components/Card/CardHeader";
import ContentSection from './ContentSection/ContentSection'
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import FooterApp from "../../shared/FooterApp/FooterApp";
import { useDispatch, useSelector } from "react-redux";
import { DashboardContext } from "./DashboardContext";
import { getDataService } from "../../middlewares/dashboardMiddleware";
import { typesRequestErrors } from "../../constants/error.constant"; 
import './Dashboard.scss'
import ScreenError from "../../shared/ScreenError/ScreenError";
import { useHistory } from "react-router-dom";
import { setErrorDashboard } from "../../actions/dashboardAction";


const Dashboard = () => {
  
  const history = useHistory();

  const dispatch = useDispatch();

  useEffect( () => {
    window.scrollTo(0,0)
    dispatch(getDataService(history)) 
    
    return () => {
    dispatch(setErrorDashboard(typesRequestErrors.getDashboard, false, null))
    }
  }, [])  // eslint-disable-line react-hooks/exhaustive-deps

  const dashboardStore = useSelector(state => state.dashboard) 

  // const [isError,setIsError] = useState(false);

  // useEffect(() => {
  //   setIsError(dashboardStore.errors[typesRequestErrors.getDashboard].isError)
  // },[dashboardStore.errors])

  const renderSuccess = () => {
    return (
      <div className="dashboardsuccess">
        <CardHeader info={dashboardStore} />
        <ContentSection/>
      </div>
    )
  }

  // const renderError = (msg) => {
  //   return (
  //     <div style={{paddingTop:"40px"}}>
  //       <ScreenError 
  //       msg={msg}
  //       />
  //     </div>
  //   )
  // }

  return (
    <>
      <DashboardContext.Provider value={dashboardStore}>
        <HeaderApp />
        <div className="dashboard-container">
          { 
            // isError ?
            // renderError(dashboardStore.errors[typesRequestErrors.getDashboard].msg): 
            renderSuccess()
          }
        </div>
        <FooterApp />
      </DashboardContext.Provider> 
      
    </>
  );
 };

export default Dashboard;
