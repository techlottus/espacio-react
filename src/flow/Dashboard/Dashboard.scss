@import '../../App.scss';

.dashboard-container {
  width: 100%;
  @extend .headerapp-top;
  @extend .footer-bottom;

  .dashboardsuccess {
    width: 100%;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      padding-top: pixeltorem(48px);
    }
  }
}


