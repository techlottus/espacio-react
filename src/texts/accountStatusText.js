export const accountStatusText = {
  statusTitle: "Estado de cuenta y consultas de facturas",
  resumeTitle: "Resumen de adeudo",
  total: "Total",
  errorMessage:
    "Tuvimos dificultades para cargar tu información en este momento",
  emptyMessage: {
    congrats: "Felicidades, estás al día con tus pagos",
    continue: "Continúa transformando tu futuro",
  },
  images: { 
    user: "images/user.png",
    empty: "yeah.png",
  },
  btnModal: {
    delete: "Eliminar",
    cancel: "Cancelar",
  },
  btnDownload: {
    download: "Descargar Estado de cuenta",
  },
  btnInvoice: {
    checkInvoice: "Consulta tus facturas",
  },
  tableAccountStatus: [
    { title: "Clave" },
    { title: "Ciclo" },
    { title: "Periodo" },
    { title: "Adeudo" },
  ],

  periodDetailTitle: "Detalles por periodo",
  periodDetailDescription:
    "Selecciona el programa y el periodo para ver los detalles de los movimientos; se clasifican de la siguiente manera:",
  periodDetailBullets: {
    debt: "Con adeudo",
    payed: "Pagados",
    toPay: "Por pagar",
  },
  periodDetailSelects: {
    program: "Programa",
    period: "Periodo",
  },
  accountDetailPeriod: {
    title: "Periodo",
    icon: "calendar_today",
    debt: "Saldo del periodo",
  },
  accountDetailMovements: {
    title: "Movimientos",
  },
  tableInvoiceQuery: [
    { title: "Fecha de emisión" },
    { title: "Periodo" },
    { title: "Concepto" },
    { title: "Monto" },
    { title: "RFC" },
    { title: "Preview" },
    { title: "PDF" },
    { title: "XML" },
  ],
  tableRowInvoiceQuery: [
    { title: "07 Abril 2021" },
    { title: "Periodo 5" },
    { title: "Colegiatura" },
    { title: "$2405.50" },
    { title: "LRKO485767HF2" },
  ],
  accountStatusInvoiceQuery: {
    title: "Consulta de facturas",
    description:
      "Filtra por rango de fechas, selecciona el perido que deseas consultar y presiona descargar.",
  },
  btnInvoiceQuery: {
    download: "Descargar todo",
    clear: "Limpiar",
  },
  linkInvoiceQuery: "Ver más"
};
