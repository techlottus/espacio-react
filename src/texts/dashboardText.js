export const dashboardText = {
  menu: {
    aula: {
      emoji: "mortar-board",
      text: "Aula",
    },
    book: {
      emoji: "books",
      text: "Biblioteca",
    },
    helpCenter: {
      emoji: "thought-balloon",
      text: "Centro de Ayuda",
    },
    procedures: {
      emoji: "file-folder",
      text: "Trámites",
    },
    accountStatus: {
      emoji: "chart-with-upwards-trend",
      text: "Estado de Cuenta",
    },
    payment:  {
      emoji: "credit-card",
      text: "Pagos",
    },
    invoice: {
      emoji: "page-facing-up",
      text: "Facturas"
    },
    benefits: {
      emoji: "star",
      text: "Portal de Beneficios"
    },
    sede: {
      emoji: "notebook",
      text: "SeDe"
    },
    psychopedagogical: {
      emoji: "blue-heart",
      text: "Centro de Atención Psicopedagógica"
    },
    evaluateServices: {
      emoji: "ec ec-white-check-mark",
      text: "Evalúa los servicios"
    },
    registrationSubjects: {
      emoji: "ec ec-closed-book",
      text: "Inscribe tus materias"
    },
    logout: {
      emoji: "back",
      text: "Cerrar sesión",
    },
  },
  titleWelcome: "Hola",
  images: {
    cardHeader: "images/header.png",
    empty: "empty.png",
  },
  logo: {
    urlLogo: "logo.svg"
  },
  logos: "logo.svg",
  sectionDashboard: {
    titleNews: "Noticias de tu universidad",
    titleCards: "¿Qué quieres hacer hoy?",
  },
  playIcon: "play_circle_outline",
  cards: {
    one: {
      title: "Aula",
      icon: "school",
    },
    two: {
      title: "Pagos",
      icon: "credit_card",
    },
    three: {
      title: "Biblioteca",
      icon: "import_contacts",
    },
    four: {
      title: "Trámites",
      icon: "post_add",
    },
    five: {
      title: "Inscribe tus materias",
      icon: "add_circle_outline",
    },
  },
  footerText: {
    text: "©2022 Derechos reservados Lottus Education",
    privacy: "Aviso de privacidad y términos y condiciones",
  },
};