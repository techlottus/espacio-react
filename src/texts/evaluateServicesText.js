export const evaluateServicesText = {
    evaluateServicesTitle: "Evalua tus servicios digitales",
    modalSurvey: {
        title: "Evalua tu experiencia para realizar tu pago a través de Espacio UTC",
        text: "¿Que tan probable es que recomiendes los pagos en Espacio UTC?"
    }
}