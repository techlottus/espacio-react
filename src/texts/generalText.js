export const generalText = {
  modals: {
    modalExit: {
      icon: "help",
      modalTitlePayment: "¿Seguro que deseas salir de la sección de pagos?",
      modalTitleProcedure: "¿Seguro que deseas abandonar el trámite?",
      exitBtn: "Aceptar",
      cancelBtn: "Cancelar",
    },
  },
};
