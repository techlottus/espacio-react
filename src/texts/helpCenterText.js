export const helpCenterText = {
  titleDashboard: "¿Cómo te podemos ayudar?",
  images: {
    headerHelpCenter: "images/back-texture-white.png",
    helpCenterFlowBanner: "images/back-texture-white.png",
    helpCenterFlowAcademic: "images/banner-ayuda-academica.png",
    helpCenterFlowFinancial: "images/banner-ayuda-financiera.png",
    helpCenterFlowInstalations: "images/banner-ayuda-instalaciones.png",
    helpCenterFlowPsychopedagogical: "images/banner-ayuda-psicometrica.png",
    helpCenterFlowTechnical: "images/banner-ayuda-tecnica.png",
    helpCenterFlowProcedures: "images/banner-ayuda-tramites.png",
    helpCenterFaqs: "images/banner-preguntas-frecuentes.png",
  },
  faqsAccordion: [
    {
      title: "¿Cuándo inician mis clases?",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos:",
      list: [
        "Partida o acta de nacimiento apostillada.",
        "Certificado o notas de bachillerato apostillado, donde deben señalar los grados y ciclos escolares cursados.",
        "Diploma o título del nivel bachillerato.",
        "Notas o analítico de secundaria o educación básica",
        "Diploma o título del nivel secundaria",
        "Identificación oficial (Si el alumno es menor de edad, deberá presentar identificación oficial del padre o tutor.",
      ],
    },
    {
      title: "¿Cuál navegador es el indicado para acceder la aula virtual ? ",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
    {
      title:
        "¿Hay una fecha límite para entregar mis actividades?, ¿Dónde puedo consultarla? ",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
    {
      title: "Tuve un problema con un profesor ¿Qué puedo hacer?",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
    {
      title: "¿Dónde puedo consultar mis calificaciones y fechas de entrega?",
      description:
        "Deberás escanear completo, a color y por ambos lados los siguientes documentos",
      list: [
        "Acta de nacimiento.",
        "Certificado parcial del nivel superior, donde deben señalar los grados y ciclos escolares cursados.",
        "Certificado total de estudios de nivel anterior inmediato.",
        "Título de nivel anterior inmediato.",
        "Identificación oficial.",
        "Carta poder.",
      ],
    },
  ],
  titleFlowFaqs: "Preguntas frecuentes",
  titleFlowTickets: "Levanta un ticket",
  arrowFormHelpCenter: {
    icon: "arrow_back",
    text: "Atrás",
  },
  modalConfirmHelpCenter: {
    success: {
      icon: "check_circle_outline",
      title: "El ticket se generó correctamente",
      firstDescription: "Tu solicitud ha sido recibida y se ha creado el ticket",
      secondDescription: "Debes conservar el No. de ticket para poder consultar el seguimiento del mismo.",
      btns: {
        next: "Ir a centro de ayuda",
      },
    },
    error: {
      img: "empty-error.png",
      title: "Ha ocurrido un error, inténtalo más tarde",
    },
  },
  modalHelpCenter: {
    title: "Verifica que la información esté correcta",
    issue: "Asunto: ",
    assignment: "Materia: ",
    professor: "Profesor: ",
    ades: "Asesor: ",
    description: "Descripción del caso: ",
    activity: "Actividad: ",
    assignmentToReport: "Material a reportar: ",
    paymentDate: 'Fecha de pago: ',
    paymentMount: 'Monto de pago: ',
    paymentMethod: 'Medio de pago: ',
    classDate: 'Fecha de la clase: '
  },
  searchResults: {
    empty: "No se han encontrado resultados para tu búsqueda",
    error: "Ha ocurrido un error, inténtalo más tarde",
  },
  directory: {
    title: "Directorio UTC",
    tableTitleOne: "Gerentes y coordinadores",
    tableTitleTwo: "Asesores educativos",
    headerTableCoordinator: [
      { title: "Nombre" },
      { title: "Área" },
      { title: "Teléfono" },
      { title: "Correo" },
    ],
    headerTableAdvisor: [
      { title: "Nombre" },
      { title: "Área" },
      { title: "Contacto" },
    ],
  },
};
