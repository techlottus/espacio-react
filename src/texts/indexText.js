import { termsText } from "./termsText";
import { paymentText } from "./paymentText";
import { dashboardText } from "./dashboardText";
import { accountStatusText } from "./accountStatusText";
import { loginText } from "./loginText";
import { screenText } from "./screenText";
import { sidebarText } from "./sidebarText";
import { proceduresText } from "./proceduresText";
import { generalText } from "./generalText";
import { helpCenterText } from "./helpCenterText";
import { evaluateServicesText } from "./evaluateServicesText";
import { registrationSubjectsText } from "./registrationSubjectsText";

export const texts = {
  terms: termsText,
  payment: paymentText,
  dashboard: dashboardText,
  accountStatus: accountStatusText,
  login: loginText,
  screen: screenText,
  sidebar: sidebarText,
  procedures: proceduresText,
  general: generalText,
  helpCenter: helpCenterText,
  evaluateServices: evaluateServicesText,
  registrationSubjects: registrationSubjectsText
}