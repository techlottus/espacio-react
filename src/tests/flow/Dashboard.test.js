import React from 'react';
import { mount } from 'enzyme';
import Dashboard from '../../flow/Dashboard/Dashboard';
import { Provider, useDispatch } from 'react-redux';
import '@testing-library/jest-dom';
import configureStore  from "redux-mock-store";
import thunk from "redux-thunk";
import { typesRequestErrors } from '../../constants/error.constant';
import { renderHook } from '@testing-library/react-hooks';
import { dashboardText } from '../../texts/dashboardText';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const mockUseLocationValue = {
  pathname: "/testroute",
  search: '',
  hash: '',
  state: null
}
jest.mock('react-router', () => ({
  ...jest.requireActual("react-router"),
  useLocation: jest.fn().mockImplementation(() => {
      return mockUseLocationValue;
  })
}));

let initState = {
  dashboard: {
    data: {
      qoutes: [
        {
          qoute: 'Hola Mund0'
        }
      ]
    },
    errors: {
      [typesRequestErrors.getDashboard]: {
        isError: false,
        msg: null
      }
    }
  },
  texts: {
    dashboard: {
      ...dashboardText
    }
  }
};

let store = mockStore(initState);

const { result } = renderHook(() => useDispatch())

const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Dashboard  />
    </Provider>
  )
}

describe('Test <Dashboard />',() => {

  beforeEach(() => {
    
  })

  test('should init component',() => {
    const wrapper = handleMount(initState);
    expect(wrapper.find('.dashboardsuccess').exists()).toBe(true)
    expect(wrapper).toMatchSnapshot();
  })

  test('should show error',() => {
    initState = {
      ...initState,
      dashboard: {
        ...initState.dashboard,
        errors: {
          ...initState.dashboard.errors,
          [typesRequestErrors.getDashboard]: {
            isError: true,
            msg: 'Error'
          }
        }
      }
    }
    const wrapper = handleMount(initState);
    expect(wrapper.find('ScreenError').exists()).toBe(true);
  })

})