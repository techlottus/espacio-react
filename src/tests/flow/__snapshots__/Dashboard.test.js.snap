// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Dashboard /> should init component 1`] = `
Array [
  <div
    className="navbar"
  >
    <lottus-navbar />
  </div>,
  <div
    className="dashboard-container"
  >
    <div
      className="dashboardsuccess"
    >
      <div
        className="cardheadercontainer"
      >
        <div
          className="cardheadercontent"
        >
          <h2
            className="cardheadergreeting"
          />
          <span
            className="cardheaderenroll"
          />
          <span
            className="cardheaderquote"
          />
        </div>
        <img
          alt="logo"
          className="cardheaderpicture"
        />
      </div>
      <div
        className="content-container"
      >
        <div
          className="contentcarousel"
        >
          <h2
            className="carouselnews"
          />
          <div
            className="carousel"
          >
            <lottus-carousel />
          </div>
        </div>
        <div
          className="contentcards"
        >
          <h2
            className="cardstitle"
          />
          <div
            className="cardsdash-container"
          >
            <div
              className="card-dash"
              key="0"
            >
              <lottus-card-dash
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <lottus-button
                  button={true}
                />
              </lottus-card-dash>
            </div>
            <div
              className="card-dash"
              key="1"
            >
              <lottus-card-dash
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <lottus-button
                  button={true}
                />
              </lottus-card-dash>
            </div>
            <div
              className="card-dash"
              key="2"
            >
              <lottus-card-dash
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <lottus-button
                  button={true}
                />
              </lottus-card-dash>
            </div>
            <div
              className="card-dash"
              key="3"
            >
              <lottus-card-dash
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <lottus-button
                  button={true}
                />
              </lottus-card-dash>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="footer"
    style={
      Object {
        "bottom": "0",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <lottus-footer />
  </div>,
]
`;
